sycl/test-e2e/OptionalKernelFeatures/no-fp64-optimization-declared-aspects.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: %{build} -o %t.out -O3
```

Comment:
Is fp64 excluded by default?

Reply:
Fp64 is handled slightly differently compared to the rest to allow some common cases, the most prominent being lowering of double-precision literals. It was admittedly a miss that explicitly marked cases should not be excluded, even if optimized out, which is what I believe this patch intends to fix.

Reply:
@KornevNikita, see [Pre- and post-optimization aspect propagation](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/OptionalDeviceFeatures.md#pre--and-post-optimization-aspect-propagation)